kore/src/Kore/Step/Step.hs

```diff
@@ -227,8 +228,9 @@ unifyRule
     -- configuration.
     let
         RulePattern { left = ruleLeft } = rule'
-    unification <- unifyPatterns ruleLeft initialTerm
-    -- Combine the unification solution with the rule's requirement clause.
+    unification <- unifyPatterns (withoutTerm initial) ruleLeft initialTerm
+    -- Combine the unification solution with the rule's requirement clause,
+    -- unless it's rendundant to do so.
```

Comment:
This looks like a merge conflict; the redundancy check was removed below. Could you fix the comment?